app.js

```diff
@@ -102,7 +102,7 @@ app.use(function(req, res, next) {
       return next();
     }
     // this belongs to one of the predefined urls, let's generate its associated page
-    if ( urls.indexOf(location) != -1 ) {
+    if ( urls.indexOf(decodeURIComponent(location)) != -1 ) {
```

Comment:
I'm not so sure about this, but urls are written in decoded human readable strings, so it makes some sense to ensure we only compare against decoded human readable string too.